File: 2020 lipiec/c++/2.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

string ident[200];

bool is_pal(const string& s)
{
    for (int i = 0; i < s.size() / 2; i++)
        if (s[i] != s[s.size() - 1 - i])
            return false;
    return true;
}

int main()
{
    ifstream in("identyfikator.txt");
    ofstream out("wyniki4_2.txt");
    
    for (int i = 0; i < 200; i++)
        in >> ident[i];
    
    for (int i = 0; i < 200; i++)
    {
        if (is_pal(ident[i].substr(0, 3)) || is_pal(ident[i].substr(3)))
            out << ident[i] << '\n';
    }
}

File: 2020 lipiec/python/2.py
with open("identyfikator.txt", "r") as f:
    ident = f.read().strip().split('\n')

with open("wyniki4_2.txt", "w") as f:
    for id in ident:
        letters = id[:3]
        numbers = id[3:]
        if letters == letters[::-1] or numbers == numbers[::-1]:
            f.write(id + '\n')